### PR TITLE
Allow the use of --user for docker containers

### LINKED ISSUE #345: Cannot use this image with a custom uid/gid and bind-mounts.
Docker allows telling a container as which user to run a process, using the `--user 1002:1002` argument or equivalent setting in docker-compose. This is especially important for docker hosts where the default uid/gid (1000:1000) belong to a user who should not have access to the data.

The official postgres docker image documentation suggests that passing this user flag and adding a read-only mapping of the hosts /etc/passwd will allow using a custom user. This does not work in timescale, as timescale has hardcoded the `postgres` user with the uid/gid 1000:1000.
https://github.com/timescale/timescaledb-docker-ha/blob/cc0deb86fcab07c25da7ac90af45ee0ff868ae62/Dockerfile#L39
https://github.com/timescale/timescaledb-docker-ha/blob/cc0deb86fcab07c25da7ac90af45ee0ff868ae62/Dockerfile#L519

Please provide guidance of bind-mounting the pgdata directory as another uid/gid than 1000:1000.

related issues:
- #272

### PR DESCRIPTION
**Pull request by graveland**
This needs to be combined with a volume mount if the user isn't uid/gid 1000/1000, as the user won't have permission to modify the default PGDATA location.

Use something like:
```
docker run \
    --user 1234:4567 \
    -e PGDATA=/home/pgdata \
    -v`pwd`/pgdata:/home/pgdata \
    $IMAGE
```

Fixed #345, and (partly?) #272. With this, #311 could be addressed potentially by specifying a path different than in that ticket, and might help a bit with #290 as well?

### CODE REVIEW

**Comment by fsiler** [comment]
+1 for this patch.

File: Dockerfile [hunk 1]
```diff
@@ -195,7 +195,7 @@ RUN cd /build && git clone https://github.com/timescale/timescaledb-docker && cd
 RUN cp -a /build/timescaledb-docker/docker-entrypoint-initdb.d /docker-entrypoint-initdb.d/
 # Add custom entrypoint to install timescaledb_toolkit
 COPY scripts/010_install_timescaledb_toolkit.sh /docker-entrypoint-initdb.d/
-RUN curl -s -o /usr/local/bin/docker-entrypoint.sh https://raw.githubusercontent.com/docker-library/postgres/${GITHUB_DOCKERLIB_POSTGRES_REF}/13/alpine/docker-entrypoint.sh
+COPY docker-entrypoint.sh /usr/local/bin/docker-entrypoint.sh
```
Comment (feikesteenbergen) [inline Dockerfile:198]: :+1:

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,377 @@
+#!/usr/bin/env bash
```
Comment (feikesteenbergen) [inline docker-entrypoint.sh:1]: Shall we add a permalink at the top to clearly show where we git it from?

Comment (feikesteenbergen) [reply docker-entrypoint.sh:1]: `s/git/get/`

**Review by feikesteenbergen — APPROVED** [review]

File: docker-entrypoint.sh (hunk 2 above)
Comment (graveland) [reply docker-entrypoint.sh:1]: Absolutely- good idea

**Comment by fsiler** [comment]
Love that this got merged- can someone help me with when this will be used to publish official images/which versions it will be available in?

**Comment by graveland** [comment]
Just today! Check out the pg15-latest, or pg14-latest (or the others from today as is appropriate: [timescaledb-ha](https://hub.docker.com/r/timescale/timescaledb-ha/tags?page=1))

**Pull request merged**